### PR TITLE
Continue the TaskScheduler items:

### PR DESCRIPTION
**Pull request by rassilon**
* Replace IWebBrowser.ZoomLevel property with:
   GetZoomLevelAsync(), and SetZoomLevel().

### CODE REVIEW

File: CefSharp.Wpf/ChromiumWebBrowser.cs [hunk 1]
```diff
@@ -506,6 +506,21 @@ private static void OnIsBrowserInitializedChanged(DependencyObject d, Dependency
...
         protected virtual void OnIsBrowserInitializedChanged(bool oldValue, bool newValue)
         {
+            managedCefBrowserAdapter.GetZoomLevelAsync()
+                .ContinueWith(new Action<Task<double>>((Task<double> previous) =>
+                    {
+                        if (previous.IsCompleted)
+                        {
+                            UiThreadRunAsync(() =>
+                            {
+                                SetValue(ZoomLevelProperty, previous.Result);
```
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: `SetCurrentValue` is needed here otherwise it will overwrite the users bindings.

Comment (rassilon) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: Done.

Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: :+1:

**Comment by amaitland** [comment]
I tweaked the `IWebBrowser` documentation slightly see https://github.com/cefsharp/CefSharp/commit/05d7860cb2a09939f7a58503815c04f2c0c74c73

File: CefSharp.Wpf/ChromiumWebBrowser.cs [hunk 2]
```diff
@@ -506,6 +506,21 @@ private static void OnIsBrowserInitializedChanged(DependencyObject d, Dependency
...
         protected virtual void OnIsBrowserInitializedChanged(bool oldValue, bool newValue)
         {
+            managedCefBrowserAdapter.GetZoomLevelAsync()
+                .ContinueWith(new Action<Task<double>>((Task<double> previous) =>
+                    {
+                        if (previous.IsCompleted)
+                        {
+                            UiThreadRunAsync(() =>
+                            {
+                                SetCurrentValue(ZoomLevelProperty, previous.Result);
```
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: Is this new code block required? As it's only called once, would the value be anything other than `0`?

Comment (rassilon) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: Paranoia,  it's chromium there might be a command line option.  :smile:

**Comment by rassilon** [comment]
Comment update looks good

File: CefSharp.Wpf/ChromiumWebBrowser.cs (hunk 1 above)
Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: I did have a quick look to see what command line options there were, very few referencing `zoom`.

`--force-device-scale-factor` is the only other that I'm aware of and it doesn't appear to change the zoom level.

http://peter.sh/experiments/chromium-command-line-switches/

Comment (rassilon) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: Or something else now or in the future. I didn't mean to be literal, it is just defensive coding.

Comment (amaitland) [inline CefSharp.Wpf/ChromiumWebBrowser.cs]: This to me falls under `YAGNI` :stuck_out_tongue: 

Leave it for now though :smile:

**Pull request merged**